src/Workspaces/Core/Portable/Options/EditorConfig/EditorConfigDocumentOptionsProviderFactory.cs

```diff
@@ -34,7 +34,7 @@ public DocumentOptions(ImmutableDictionary<string, string> options)
 
                 public bool TryGetDocumentOption(OptionKey option, out object? value)
                 {
-                    var editorConfigPersistence = option.Option.StorageLocations.OfType<IEditorConfigStorageLocation>().SingleOrDefault();
+                    var editorConfigPersistence = (IEditorConfigStorageLocation?)option.Option.StorageLocations.SingleOrDefault(static location => location is IEditorConfigStorageLocation);
```

Comment:
Is this just better because it's not allocating another enumerable, or is OfType funky in some other way? (just curious)

Reply:
It skips two enumerable allocations. One for OfType, and the other because ImmutableArray provides a non-boxing overload of SingleOrDefault.